R/param-estimates.R

```diff
@@ -105,3 +108,71 @@ is_diag <- function(.name) {
 
   return(.ind[1] == .ind[2])
 }
+
+
+#' Add shrinkage details to param table
+#'
+#' Private helper to unpack shrinkage details and create a column on the input tibble
+#' with the shrinkage assigned to the relevant parameters.
+#' @param .param_df The parameter estimates table containing, at minimum `parameter_names` and `diag` columns.
+#' @param .summary A `bbi_nonmem_summary` object.
+#' @importFrom dplyr select filter left_join bind_rows
+#' @importFrom stringr str_detect
+#' @importFrom tibble add_column
+#' @keywords internal
+add_param_shrinkage <- function(.param_df, .summary) {
+
+  # extract shrinkage for final estimation method
+  shk <- .summary[[SUMMARY_SHRINKAGE]]
+  shk <- shk[[length(shk)]]
+
+  if (is.null(shk)) {
+    return(.param_df %>% add_column(!!SUMMARY_PARAM_SHRINKAGE := NA_real_))
```

Comment:
1. getting rid of this per previous comment because don't need `add_column`

2. I see the value here but I'm gonna pass because we want to move this release forward. Plus, I generally like the "left-justified" control flow mentality. Topic for later discussion perhaps.